### DISCUSSION TITLE
Cruise control pod restarting with Cannot read field "brokerCapacities"

### DISCUSSION THREAD

**Discussion by sgv0007** [strimzi/strimzi-kafka-operator#10897] [Q&A] [answered]
Hi Team ,

I recently upgraded strimzi to 0.42 from 0.38 and I noticed below error in cruise control pod logs. I see capacity.json file is present in the Config Maps under [kafka-broker-cruise-control-config].Can you please let me know what is the fix for this error.

Error:
2024-12-02T11:53:09.379947756Z 2024-12-02T11:53:09.377Z ERROR --- [ main] KafkaCruiseControlMain:33 - Uncaught exception on thread Thread[main,5,main]
2024-12-02T11:53:09.379971349Z java.lang.NullPointerException: Cannot read field "brokerCapacities" because the return value of "com.google.gson.Gson.fromJson(com.google.gson.stream.JsonReader, java.lang.reflect.Type)" is null
2024-12-02T11:53:09.379976607Z 	at com.linkedin.kafka.cruisecontrol.config.BrokerCapacityConfigFileResolver.loadCapacities(BrokerCapacityConfigFileResolver.java:298) ~[cruise-control-2.5.128.jar:?]
2024-12-02T11:53:09.379979514Z 	at com.linkedin.kafka.cruisecontrol.config.BrokerCapacityConfigFileResolver.configure(BrokerCapacityConfigFileResolver.java:161) ~[cruise-control-2.5.128.jar:?]

Cruise control Config in kafka crd-

    cruiseControl:
      image: abc.com/strimzi/kafka:0.38.0-kafka-3.6.0       
    metricsConfig:
      type: jmxPrometheusExporter
      valueFrom:
        configMapKeyRef:
          name: cruise-control-metrics
          key: metrics-config.yml
    resources:
      requests:
        cpu: 200m
        memory: 512Mi
      limits:
        cpu: 1
        memory: 2Gi
    logging:
        type: external
        valueFrom:
          configMapKeyRef:
            name: eesb-kafka-logging-config
            key: cruise-control-log4j2.properties
    template:
      cruiseControlContainer:
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop:
              - ALL
      pod:
        metadata:
          labels:
          annotations:

        securityContext:
          runAsNonRoot: true
          seccompProfile:
            type: RuntimeDefault

**Comment by scholzj** [comment, marked as answer]
If you use Strimzi 0.42.0, you should not use the image `abc.com/strimzi/kafka:0.38.0-kafka-3.6.0` from 0.38. You should use the 0.42 Cruise Control image. This can cause issues like this, so the first step would be to fix this issue I guess.

**Reply by scholzj** [reply]
As a side note ... not sure if this is just some copy paste issue ... but the YAML seems completely misaligned and most of it is not in the `cruiseControl` section.

**Reply by sgv0007** [reply]
@scholzj Thanks for the reply, Since 0.42 supports kafka 3.6.0 I kept the same image . I will update the image and check again .

Yes the Yaml misalignment is due to copy paste issue .

**Reply by scholzj** [reply]
There is more stuff in the image than just Kafka. The Cruise Control version might be completely different, there are various startup scripts etc. So you always need to use the images for the Strimzi version you use.
